### DISCUSSION TITLE
IQuery.With() throws exception when provided with a more complex query.

### DISCUSSION THREAD

**Discussion by emil-aleksandrov** [OrchardCMS/OrchardCore#10390] [General]
Using IQuery's With() or Where() methods you get a "'System.InvalidOperationException' in System.Linq.Expressions.dll" when you provide the method with the expression in the section below.

### Actual behavior
```
var asd = new List<string> { "a", "s", "d" };

var query = _session.Query<ContentItemIndex, ContentItemIndex>();
query.With<ContentItemIndex>(x => asd.Any(z => x.DisplayText.Contains(z)));
```

On the 3rd row, you will get an exception - "System.InvalidOperationException' in System.Linq.Expressions.dll"

### Expected behavior
Notice that this is not a problem with Linq itself since it works in EF for example.

```
var asd = new List<string> {"a","s","d" };

var query = _repository.Where(x => asd.Any(z => x.Name.Contains(z)));
```

If anyone knows any workarounds to this, I will be very grateful to read them because this is a blocker for our project right now.

**Comment by deanmarcussen** [comment]
You need to use the YesSql extensions.

Namespace is `using YesSql.Services` and you probably want `.Where(x -> x.Name.IsIn(asd));`
